### PR TITLE
Update rubocop dependencies to latest versions

### PR DESCRIPTION
**Pull request by ticklemynausea**
* Upgrade `rubocop` to v1.42.0.
* Upgrade `rubocop-performance` to v1.15.2.
* Upgrade `rubocop-rails` to v2.17.4.
* Upgrade `rubocop-rspec` to v2.16.0.

### CODE REVIEW

File: lib/salsify_rubocop/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module SalsifyRubocop
-  VERSION = '1.27.1'
+  VERSION = '1.3.0'
```
Comment (erikkessler1) [inline lib/salsify_rubocop/version.rb]: As per our [versioning conventions](https://github.com/salsify/salsify_rubocop#versioning) for this gem, this version should be `1.42.0` to match the version of rubocop.

**Review by erikkessler1 — COMMENTED** [review]
Tested it out in content-flow-service and there were ~40 violations most of which were autocorrectable and the others looked easy to fix, so this seems reasonable to me.

I did notice this warning though:

```
Warning: obsolete parameter `IgnoredPatterns` (for `Layout/LineLength`) found in /Users/ekessler/salsify_rubocop/conf/rubocop_without_rspec.yml
`IgnoredPatterns` has been renamed to `AllowedPatterns`.
```

I think we need to update this line: https://github.com/salsify/salsify_rubocop/blob/b5dff76308a46678e11216daa63efb4ffad8a79e/conf/rubocop_without_rspec.yml#L182

File: lib/salsify_rubocop/version.rb (hunk 1 above)
Comment (ticklemynausea) [reply lib/salsify_rubocop/version.rb]: Missed that. Apologies.

**Comment by ticklemynausea** [comment]
I need this for `con-u` , and got precisely 33 autocorrectable offenses.

**Review by erikkessler1 — APPROVED** [review]

**Pull request merged**